File: app/src/main/java/ml/puredark/hviewer/dataproviders/AbstractDataProvider.java

package ml.puredark.hviewer.dataproviders;

import java.util.Collection;
import java.util.List;

public abstract class AbstractDataProvider {

    public static abstract class Data {
        public abstract int getId();
    }

    public abstract int getCount();

    public abstract List getItems();

    public abstract Data getItem(int index);

    public abstract void removeItem(int position);

    public abstract void clear();

    public abstract void addItem(Data item);

    public abstract void addItem(int position, Data item);

    public abstract void addAll(Collection items);
}

File: app/src/main/java/ml/puredark/hviewer/dataproviders/AbstractDataProvider.kt
package ml.puredark.hviewer.dataproviders

abstract class AbstractDataProvider {

    abstract class Data {
        abstract val id: Int
    }

    abstract val count: Int

    abstract val items: List<Any>

    abstract fun getItem(index: Int): Data

    abstract fun removeItem(position: Int)

    abstract fun clear()

    abstract fun addItem(item: Data)

    abstract fun addItem(position: Int, item: Data)

    abstract fun addAll(items: Collection<Any>)
}
